### ISSUE TITLE
useUpdateFactory hook not listening to props changing

### ISSUE THREAD

**Issue by charlynmanuyag** [snowcoders/react-context-store#63]
In my application, I have this snippet of code where essentially I am using modifiers useUpdateFactory hook to make an API call using some id coming from props. Note that I have only tested this on useUpdateFactory and not the other hooks

**Expected behavior:**
When id changes from id1 to id2 when calling getAll, id should be equal to id2

**Actual behavior:**
When id changes from id1 to id2, when calling getAll, id is still equal to id1

```
export function ApiProvider(props: PropsWithChildren<{id: string}>) {
  const { id, children } = props;
  const [state, modifiers] = useIndexableContextStore(defaultValue);

  const getAll = modifiers.useUpdateFactory({
    action: async () => {
      console.log('id', id)
      const value = await getSomeValue({
        id,
      });
      return value;
    },
  });

  return <Context.Provider value={{ ...state, getAll }}>{children}</Context.Provider>;
}
```

**Comment by k2snowman69** [comment]
I believe this is due to useUpdateFactory using a useCallback but not providing a way to define dependencies that would force the useCallback to update.

Let me verify this assumption, hopefully should have something within a day or two.

**Comment by k2snowman69** [comment]
I've released react-context-store@1.1.0-beta.0 . If it fixes your issue, I can publish a non-beta build for you, let me know how it goes!

**Comment by charlynmanuyag** [comment]
Validated with react-context-store@1.1.0-beta.0 by passing a dependency array and it's working as expected.

**Comment by k2snowman69** [comment]
Fixed with react-context-store@1.1.0
